Move new features to Highlight. Add additionalRouters and gRPC (#1779)
lagom/lagom: docs/manual/java/releases/Highlights15.md

# What's new in Lagom 1.5?

This page highlights the new features of Lagom 1.5. If you want to learn about the changes you need to make when you migrate to Lagom 1.5, check out the [[Lagom 1.5 Migration Guide|Migration15]].

## TLS Support

> Lagom in both [[dev mode|ConfiguringServicesInDevelopment#Using-HTTPS-in-development-mode]] and [[tests|Test#How-to-use-TLS-on-tests]] supports basic usage of TLS by means of self-signed certificates provided by the framework.

Author: The different page name of `Test` in `java` vs `TestingServices` in `scala` is the only difference between the `Highlights15.md` files.

Just because of this link I had to duplicate `Highlights15.md`.

Reviewer: That sucks.  Is this easily fixable?

Author: Only option is to remove the link.

Author: This is a page we'll barely ever come back to.

Reviewer 2: Instead of duplicating the highlights file, I would prefer to do something like: "see the docs for Java or Scala" and have different links. Is this possible?

Author: This is not possible.

I think the `Highlights15.md` is a page we won't be editing anymore. We already have a ton of duplicated pages that need more maintenance. Once `1.5.0` is out, `Highlights15.md` will very rarely be edited.

Reviewer: Implemented in https://github.com/lagom/lagom/pull/1788.

## Cluster Formation

A new mechanism to form and [[join an Akka Cluster|Cluster#Joining]] is introduced in Lagom 1.5. Apart from the original `Manual Cluster Formation` the new `Akka Cluster Bootstrap` is now supported. This new mechanism is introduced with lower precedence than `Manual Cluster Formation` so if you rely on the use of a list of `seed-nodes` then everything will work as before. On the other hand, `Akka Cluster Bootstrap` takes precedence over the `join-self` cluster formation for single-node clusters. If you use single-node clusters via `join-self` you will have to explicitly disable `Akka Cluster Bootstrap`:

```
lagom.cluster.bootstrap.enabled = false
```

##  Additional Routers

> Starting in Lagom 1.5 it is possible to extend the routes exposed by your service. So not only your service will expose thhe calls listed on  `Service.Descriptor` but can also serve endpoints handled by `additionalRouters`. The [[documentation|AdditionalRouters]] covers all the details. Additional routers makes it trivial to extend a `Service.Descriptor` with features natively supported by Play such has Upload File. There's even [Lagom recipes](https://github.com/lagom/lagom-recipes/tree/master/file-upload) detailing such a use case.

Reviewer: ```suggestion
As of Lagom 1.5 it is possible to extend the routes exposed by your service. So your service will not only expose the calls listed in `Service.Descriptor` but will also serve endpoints handled by `additionalRouters`. The [[documentation|AdditionalRouters]] covers all the details. Additional routers make it trivial to extend a `Service.Descriptor` with features natively supported by Play such as Upload File. There is also a [Lagom recipe](https://github.com/lagom/lagom-recipes/tree/master/file-upload) detailing such a use case.
```

> Starting in Lagom 1.5 it is possible to extend the routes exposed by your service. So not only your service will expose thhe calls listed on  `Service.Descriptor` but can also serve endpoints handled by `additionalRouters`. The [[documentation|AdditionalRouters]] covers all the details. Additional routers makes it trivial to extend a `Service.Descriptor` with features natively supported by Play such has Upload File. There's even [Lagom recipes](https://github.com/lagom/lagom-recipes/tree/master/file-upload) detailing such a use case.

Reviewer 3: fix "...expose thhe calls..."

```suggestion
Starting in Lagom 1.5 it is possible to extend the routes exposed by your service. So not only your service will expose the calls listed on  `Service.Descriptor` but can also serve endpoints handled by `additionalRouters`. The [[documentation|AdditionalRouters]] covers all the details. Additional routers makes it trivial to extend a `Service.Descriptor` with features natively supported by Play such as File Upload. There's even [Lagom recipes](https://github.com/lagom/lagom-recipes/tree/master/file-upload) detailing such a use case.
```

Reviewer 3: Shall we merge 
https://github.com/lagom/lagom-recipes/pull/67 and https://github.com/lagom/lagom-recipes/pull/68?

Author: Merged

Author: The change suggested by @dwijnand above superseded this suggestion.

Reviewer 3: No entirely. We should say: "File Upload", not "Upload File".

Reviewer 3: Alternatively, we could write: "with features natively supported by Play such as uploading a file"
